docs/content/en/integrations/social-authentication.md

```diff
@@ -227,14 +227,45 @@ Follow along below.
     DD_SOCIAL_AUTH_GITLAB_PROJECT_AUTO_IMPORT = True
     {{< /highlight >}}
 
-5.  Restart DefectDojo, and you should now see a **Login with Gitlab**
+5. Restart DefectDojo, and you should now see a **Login with Gitlab**
     button on the login page.
 
+## Keycloak
+1. go to your keycloak realm, add a new client of type openid-connect (use client id for DD_SOCIAL_AUTH_KEYCLOAK_KEY)
+2. in the client settings:
+   1. set access type to confidential
+   2. under valid Redirect URIs, add '<YOUR_SITE>/*'
+   3. under web origins, add the same (or '+')
+   4. fine grained openID connect configuration -> user info signed response alogrith: set to RS256
+   5. fine grained openID connect configuration -> request object signature algorithm: set to RS256
+3. Under Scope -> Full Scope Allowed -> Disable
+4. under mappers -> add custom mapper
+   * name: aud
+   * Mapper type: audience
+   * included audience: select your client/client-id here
+   * add ID to token: off
+   * add access to token: on
+5. under credentials: copy secret (use as DD_SOCIAL_AUTH_KEYCLOAK_SECRET below)
+6. in your realm settings -> keys: copy the "Public key" (signing key) (use for DD_SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY below)
+7. in your realm settings -> general -> endpoints look into openId endpoint configuration
+   and look up your authorization and token endpoint
+8. add to extraConfigs in chart:
+```
+DD_SESSION_COOKIE_SECURE: 'True'
+DD_CSRF_COOKIE_SECURE: 'True'
+DD_SECURE_SSL_REDIRECT: 'True'
+DD_SOCIAL_AUTH_KEYCLOAK_OAUTH2_ENABLED: 'True'
+DD_SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY: 'MIIBWGn....zwIDAQAB'
```

Comment:
Another way around is to fetch key from `https://keycloak-server/auth/realms/my-realm` maybe this can be improvement to automate configuration.

Reply:
Indeed an interesting suggestion. But can we really do so? Looking at the [social auth plugin documentation section for keycloak](https://github.com/python-social-auth/social-core/blob/master/social_core/backends/keycloak.py), I don't really see that option. In case you know more, let me know!

Reply:
You are right, there is not direct integration. I'm saying this because pushing pub key in vars can be tricky (special characters) or simple typo.